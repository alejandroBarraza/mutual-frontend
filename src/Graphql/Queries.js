import { gql } from '@apollo/client';

//videos queries(always get last created)
export const GETVIDEOS = gql`
    query Getvideo {
        videos(limit: 1, sort: "created_at:desc") {
            created_at
            titulo
            descripcion
            links
        }
    }
`;

//statistic query(always get last 4 created)
export const GETSTATISTICS = gql`
    query GetStatistic {
        statistics(limit: 4, sort: "created_at:asc") {
            titulo
            descripcion
            created_at
        }
    }
`;

// get download content given limit and start =="offset" as arguments
export const GETDOWNLOADS_LIMIT_START = gql`
    query GetDownloads($start: Int!, $limit: Int!) {
        downloads(start: $start, limit: $limit) {
            id
            published_at
            titulo
            imagenes {
                name
                url
            }
        }
    }
`;

export const GETDOWNLOADS_SORTED_DESC = gql`
    query GetDownloads {
        downloads(sort: "created_at:desc") {
            id
            titulo
            descripcion
            archivo {
                url
            }
        }
    }
`;

export const GETNEWS_START_LIMIT = gql`
    query getNews($limit: Int!, $start: Int!) {
        news(limit: $limit, start: $start, sort: "created_at:desc") {
            id
            titulo
            descripcion
            tiempoLectura
            imagen {
                url
            }
        }
    }
`;

// get the footer elements
export const GET_ALL_FOOTER = gql`
    query getFooter {
        footer {
            selloSeguridad
            descripcion
            instagram
            facebook
            twitter
            youtube
        }
    }
`;
