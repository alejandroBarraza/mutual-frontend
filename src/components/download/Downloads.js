import React from 'react';
import { styled } from '@mui/material/styles';
import Card from '@mui/material/Card';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';

const ExpandMore = styled((props) => {
    const { expand, ...other } = props;
    return <IconButton {...other} />;
})(({ theme, expand }) => ({
    transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
        duration: theme.transitions.duration.shortest,
    }),
}));

const CustomCss = {
    card: { maxWidth: 345, margin: 'auto' },
    imagenCustom: {
        filter: 'brightness(70%)',
        transition: 'filter 0.5s ease-in-out',
        '&:hover': {
            filter: 'none',
        },
    },
};

export const Downloads = () => {
    const [expanded, setExpanded] = React.useState(false);

    const handleExpandClick = () => {
        setExpanded(!expanded);
    };

    return (
        <div className='container'>
            <Card sx={CustomCss.card}>
                <CardMedia
                    component='img'
                    height='194'
                    image='https://image.freepik.com/vector-gratis/fondo-halloween-plano-dibujado-mano_23-2149109803.jpg'
                    alt='random imagen'
                    sx={CustomCss.imagenCustom}
                />
                <CardContent>
                    <Typography variant='body2' color='text.secondary'>
                        This impressive paella is a perfect party dish and a fun meal to cook
                        together with your guests. Add 1 cup of frozen peas along with the mussels,
                        if you like.
                    </Typography>
                </CardContent>
                <CardActions disableSpacing>
                    <ExpandMore
                        expand={expanded}
                        onClick={handleExpandClick}
                        aria-expanded={expanded}
                        aria-label='show more'>
                        <ExpandMoreIcon />
                    </ExpandMore>
                </CardActions>
                <Collapse in={expanded} timeout='auto' unmountOnExit>
                    <CardContent>
                        <Typography paragraph>Method:</Typography>
                        <Typography paragraph>
                            Heat 1/2 cup of the broth in a pot until simmering, add saffron and set
                            aside for 10 minutes.
                        </Typography>
                        <Typography paragraph>
                            Heat oil in a (14- to 16-inch) paella pan or a large, deep skillet over
                        </Typography>
                        <Typography paragraph>
                            Add rice and stir very gently to distribute. Top with artichokes and p
                        </Typography>
                        <Typography>
                            Set aside off of the heat to let rest for 10 minutes, and then serve.
                        </Typography>
                    </CardContent>
                </Collapse>
            </Card>
        </div>
    );
};
