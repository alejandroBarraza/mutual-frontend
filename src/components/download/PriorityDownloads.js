import { useQuery } from '@apollo/client';
import { Skeleton } from '@mui/material';
import { Box } from '@mui/system';
import React from 'react'
import { GET_PRIORITY_DOWNLOADS } from '../../Graphql/Queries';
import { ActionAreaCard } from '../news/ActionAreaCard';

export const PriorityDownloads = () => {
   const { loading, error, data } = useQuery(GET_PRIORITY_DOWNLOADS);


   return (
      <Box sx={{ width: '100%' }}>
         {
            (data && data.downloads.lengh !== 0) ? data.downloads.map((element) => {
               return (
                     <Box sx={{ width: '100%', py: '.3rem' }} >
                        <ActionAreaCard download={element} />
                     </Box>)
            }) : <Skeleton variant="rectangular" width={'100%'} height={'100%'} />
         }
      </Box>
   );
}