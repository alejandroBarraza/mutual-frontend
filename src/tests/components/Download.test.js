import { render, cleanup, waitFor, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import '@wojtekmaj/enzyme-adapter-react-17';
import { GETDOWNLOADS_SORTED_DESC } from '../../Graphql/Queries';
import { Downloads } from '../../components/download/Downloads';

afterEach(cleanup);

const mocks = [
    {
        request: {
            query: GETDOWNLOADS_SORTED_DESC,
        },
        result: {
            data: {
                downloads: [
                    {
                        id: '4',
                        titulo: 'descarga de prueba',
                        descripcion: 'esta es una descarga de prueba de el dia martes.',
                        archivo: {
                            url: 'https://res.cloudinary.com/dkfgeruxd/image/upload/v1635862641/197111763_Certificado_Curricular_05_10_2021_21_47_24_45bc422a2b.pdf',
                        },
                    },
                    {
                        id: '3',
                        titulo: 'choques antofagasta 2',
                        descripcion: '*ejemplo de contenido*',
                        archivo: null,
                    },
                    {
                        id: '2',
                        titulo: 'contenido descargable 1 ',
                        descripcion: 'ejemplo de contenido de descarga 1',
                        archivo: null,
                    },
                ],
            },
        },
    },
];

const mocksError = [
    {
        request: {
            query: GETDOWNLOADS_SORTED_DESC,
        },
        error: new Error('An error occurred'),
    },
];

const mocksEmptyData = [
    {
        request: {
            query: GETDOWNLOADS_SORTED_DESC,
        },
        result: {
            data: {
                downloads: [],
            },
        },
    },
];

describe('Testing <Download/> component at error,loading,empty array and data fetched states.', () => {
    it('should render <Loading/> before graphql promise get resolved.', () => {
        const { container } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Downloads />
            </MockedProvider>
        );

        //select loading component root class.
        const loading = container.querySelector('.MuiCircularProgress-root');
        expect(loading).toBeInTheDocument();
    });

    it('should render the Error ui component in Error Fetch Data', async () => {
        const { container } = render(
            <MockedProvider mocks={mocksError} addTypename={false}>
                <Downloads />
            </MockedProvider>
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        const error = container.querySelector('.container-error');
        expect(error).toMatchSnapshot();
    });

    it('should render "<no hay contenido descargable>" when data array is empty ', async () => {
        render(
            <MockedProvider mocks={mocksEmptyData} addTypename={false}>
                <Downloads />
            </MockedProvider>
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        const element = screen.getByTestId('no-content');
        console.log(element);
        expect(element).toBeInTheDocument();
    });
});
