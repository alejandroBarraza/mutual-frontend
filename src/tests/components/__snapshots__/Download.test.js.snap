// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Download/> component at error,loading,empty array and data fetched states. should render the Error ui component in Error Fetch Data 1`] = `
<div
  class="container-error"
>
  <p>
    "An error occurred"
  </p>
</div>
`;
